src/server/scripts/ExilesReach/zone_exiles_reach.cpp

```diff
@@ -1559,77 +1623,416 @@ struct npc_horde_survivors_beach_q59930_private : public ScriptedAI
+                        }
+                        else
+                        {
+                            Conversation* conversation = Conversation::CreateConversation(CONVERSATION_LINE_ESCORT_ALLIANCE_SURVIVOR, player, *player, _playerGUID, nullptr, false);
+                            conversation->AddActor(ACTOR_ID_ALLIANCE_SURVIVOR, ACTOR_INDEX_SURVIVOR_ONE, me->GetGUID());
+                            conversation->Start();
+
+                            _events.ScheduleEvent(EVENT_FOLLOW_PLAYER, 2s);
+                        }
+                    }
+                    break;
+                case EVENT_FOLLOW_PLAYER:
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                        me->GetMotionMaster()->MoveFollow(player, 0.0f, 0.0f);
+                    break;
+                default:
+                    break;
             }
         }
     }
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+    bool _reachedCamp;
+};
 
-    void UpdateAI(uint32 /*diff*/) override
+// This script is used by Warlord Grimaxe 166805 Follower for Finding the Lost Expedition quest
+struct npc_grimaxe_summoned_beach : public EscortAI
+{
+    npc_grimaxe_summoned_beach(Creature* creature) : EscortAI(creature) {}
+
+    void IsSummonedBy(WorldObject* summoner) override
     {
-        if (!UpdateVictim())
+        if (summoner->GetTypeId() != TYPEID_PLAYER)
             return;
 
-        DoMeleeAttackIfReady();
+        Player* player = summoner->ToPlayer();
+        _playerGUID = player->GetGUID();
+
+        _reachedCamp = false;
+
+        _events.ScheduleEvent(EVENT_INITIAL_SPAWN_CHECK, 1s);
+    }
+
+    void SpellHit(WorldObject* caster, SpellInfo const* spellInfo) override
+    {
+        if (spellInfo->Id != SPELL_GARRICK_PING)
+            return;
+
+        if (_reachedCamp)
+            return;
+
+        _reachedCamp = true;
+
+        if (Player* player = caster->ToPlayer())
+        {
+            Creature* wonza = FindCreatureIgnorePhase(player, "wonza_camp", 50.0f);
+            Conversation* conversation = Conversation::CreateConversation(CONVERSATION_LINE_ESCORT_SURVIVOR_CAMP, player, *player, _playerGUID, nullptr, false);
+            conversation->AddActor(ACTOR_ID_HORDE_WONZA, ACTOR_INDEX_SURVIVOR_TWO, wonza->GetGUID());
+            conversation->AddActor(ACTOR_ID_HORDE_SURVIVOR, ACTOR_INDEX_SURVIVOR_THREE, me->GetGUID());
+            conversation->Start();
+
+            me->GetMotionMaster()->Remove(FOLLOW_MOTION_TYPE);
+            me->GetMotionMaster()->MovePoint(0, -249.20117f, -2492.6191f, 17.964903f, false);
+        }
+    }
+
+    void MovementInform(uint32 uiType, uint32 /*uiId*/) override
+    {
+        if (uiType != POINT_MOTION_TYPE)
+            return;
+
+        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+        {
+            player->CastSpell(player, SPELL_UPDATE_PHASE_SHIFT);
+            player->UpdateObjectVisibility();
```

Comment:
is this required although we are already doing manual phase update via spell?

Reply:
That is the quest phase shift being updated by spell. And it is required for this quest.

Reply:
That is required to shift phase on this quest.